eslint_rules/src/rules/dto-and-entity-style.test.ts

```diff
@@ -0,0 +1,46 @@
+import path from "node:path";
+import tseslint from "typescript-eslint";
+import { RuleTester } from "@typescript-eslint/rule-tester";
+import * as vitest from "vitest";
+
+import { rule } from "./dto-and-entity-style.js";
+
+RuleTester.afterAll = vitest.afterAll;
+RuleTester.it = vitest.it;
+RuleTester.itOnly = vitest.it.only;
+RuleTester.describe = vitest.describe;
+
+const ruleTester = new RuleTester({
+  languageOptions: {
+    parser: tseslint.parser,
+    parserOptions: {
+      projectService: {
+        allowDefaultProject: ["*.ts*"],
+        defaultProject: "tsconfig.json",
+      },
+      tsconfigRootDir: path.join(__dirname, "../.."),
+    },
+  },
+});
+
+ruleTester.run("no-loop-over-enum", rule, {
+  valid: [
+    `class FooDTO { a!: number; b?: number}`,
+    `class FooDTO { a!: number}`,
+    `class FooDTO { a?: number}`,
```

Comment:
Could we throw in a class that doesn't end with DTO like `class FooBar { a: number}`